modules/python/compat/compat-python-v3.c

```diff
@@ -32,3 +35,22 @@ py_object_as_string(PyObject *object)
 {
   return PyUnicode_AsUTF8(object);
 }
+
+void
+py_setargv(int argc, char **argv)
+{
+  GPtrArray *wchar_argv =  g_ptr_array_new_full(argc, (GDestroyNotify)g_free);
+
+  const int single_arg_len = 100;
```

Comment:
...this limit here looks very arbitrary. If the function is to be used internally, to set `argv[0]`, then this limit is too big. If it can be used by other code, then it feels too limiting. I understand that doing it otherwise is non-trivial.

But perhaps we can nuke the whole function, and use something a lot simpler, which only does the one thing we need it for.